src/stack/Stack.java

```diff
@@ -1,56 +1,83 @@
 package stack;
 
+import java.util.Arrays;
+
 public class Stack {
 
-	static final int DEFAULT_SIZE = 16;
-	private int[] data;
-	private int tos; //index which determines the top of stack
+    static final int DEFAULT_SIZE = 16;
+    private int[] data;
+    private int tos; //index which determines the top of stack
+
+
+    public Stack() {
+        this(DEFAULT_SIZE);
+    }
 
-	public Stack() {
-		this(DEFAULT_SIZE);
-	}
+    public Stack(int size) {
+        data = new int[size];
+        this.tos = -1;
+    }
 
-	public Stack(int size) {
-		data = new int[size];
-		this.tos = -1;
-	}
+    /**
+     * Add element into stack
+     */
+    public void push(int value) {
+        if (tos == data.length - 1) {
+            ensureCapacity();
+        }
+        data[++tos] = value;
+    }
 
-	/**
-	 * Add element into stack
-	 */
-	public void push(int value) {
+    /**
+     * Get element from the stack
+     *
+     * @return
+     */
+    public int pop() {
+        if (isEmpty()) {
+            System.out.print(" Stack is empty ");
+            return -1;
+        } else {
+            return data[tos--];
+        }
+    }
 
-	}
 
-	/**
-	 * Get element from the stack
-	 * @return
-	 */
-	public int pop() {
+    /**
+     * Clear stack
+     */
+    public void clear() {
 
-		return 0;
-	}
+        data = new int[DEFAULT_SIZE];
+        tos = -1;
 
+    }
 
-	/**
-	 * Clear stack
-	 */
-	public void clear() {
+    /**
+     * Check if stack is empty
+     *
+     * @return
+     */
+    public boolean isEmpty() {
+        return 0 > tos;
```

Comment:
It would be right to write like this

for (int i = 0; i < data.length; i++) {
			if (data[i] != 0){
				return false;
			}
		}
		return true;